clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -187,6 +205,18 @@ public RecordAppendResult append(TopicPartition tp,
                                      Header[] headers,
                                      Callback callback,
                                      long maxTimeToBlock) throws InterruptedException {
+        return append(tp, timestamp, key, value, headers, callback, maxTimeToBlock, null, false);
+    }
+
+    public RecordAppendResult append(TopicPartition tp,
```

Comment:
Could we add a javadoc here as well, mainly to describe the `boolean manualPartition` parameter?

Reply:
I think we should also mention that the `Cluster cluster` parameter is nullable, that sounds important and might help a future developer avoid NPEs

Reply:
Currently it is only nullable in testing situations. Should I still mention it is nullable?

Reply:
This should never be null except for tests, right? So I think we can resolve this

Reply:
The parameter has been completely removed. 😅 Very safe to resolve.